src/cli/run.rs

```diff
@@ -134,11 +135,20 @@ impl Run {
                 }
             })
             .flat_map(|args| args.split_first().map(|(t, a)| (t.clone(), a.to_vec())))
-            .map(|(t, args)| match config.tasks_with_aliases().get(&t) {
-                Some(task) => Ok(task.clone().with_args(args.to_vec())),
-                None if t == "default" => self.prompt_for_task(config),
-                None => bail!("no task {} found", style::ered(t)),
-            })
+            .map(
+                |(t, args)| match (t.as_ref(), config.tasks_with_aliases().get_matching(&t)?) {
+                    ("default", tasks) if tasks.is_empty() => {
+                        Ok(vec![self.prompt_for_task(config)?])
+                    }
+                    (_, tasks) if tasks.is_empty() => bail!("no task {} found", style::ered(t)),
+                    (_, tasks) => Ok(tasks
+                        .iter()
+                        .cloned()
+                        .map(|t| t.clone().with_args(args.to_vec()))
+                        .collect()),
+                },
+            )
```

Comment:
I think the pattern matching is getting a bit crazy here, what do you think of this?

```suggestion
            .map(|(t, args)| {
                let tasks = config.tasks_with_aliases().get_matching(&t)?;
                if tasks.is_empty() {
                    ensure!(t == "default", "no task {} found", style::ered(t));
                    Ok(vec![self.prompt_for_task(config)?])
                } else {
                    Ok(tasks
                        .iter()
                        .cloned()
                        .map(|t| t.clone().with_args(args.to_vec()))
                        .collect())
                }
            })
```

probably not shorter but I think it's easier to grok